src/providers/hana/qgshanaprovider.cpp

```diff
@@ -0,0 +1,1780 @@
+    return;
+
+  QgsRectangle ext;
+  bool isRoundEarth = false;
+  QString sql = QStringLiteral( "SELECT MIN_X, MIN_Y, MAX_X, MAX_Y, ROUND_EARTH FROM SYS.ST_SPATIAL_REFERENCE_SYSTEMS "
+                                "WHERE SRS_ID = ?" );
+  QgsHanaConnectionRef conn( mUri );
+  QgsHanaResultSetRef rs = conn->executeQuery( sql, { mSrid } );
+  if ( rs->next() )
+  {
+    ext.setXMinimum( rs->getDouble( 1 ) );
+    ext.setYMinimum( rs->getDouble( 2 ) );
+    ext.setXMaximum( rs->getDouble( 3 ) );
+    ext.setYMaximum( rs->getDouble( 4 ) );
+    isRoundEarth = ( rs->getString( 5 ) == QLatin1String( "TRUE" ) );
+  }
+  rs->close();
+  mSrsExtent = ext;
+
+  if ( isRoundEarth )
+  {
+    sql = QStringLiteral( "SELECT COUNT(*) FROM SYS.ST_SPATIAL_REFERENCE_SYSTEMS WHERE SRS_ID = ?" );
+    mHasSrsPlanarEquivalent = conn->executeCountQuery( sql, { QgsHanaUtils::toPlanarSRID( mSrid ) } ) > 0;
+  }
+}
+
+long QgsHanaProvider::getFeatureCount( const QString &whereClause ) const
+{
+  QString sql = QStringLiteral( "SELECT COUNT(*) FROM (%1)" ).arg( buildQuery( mQuery, whereClause ) );
+  QgsHanaConnectionRef conn( mUri );
+  size_t count = conn->executeCountQuery( sql );
+  return static_cast<long>( count );
+}
+
+QgsCoordinateReferenceSystem QgsHanaProvider::crs() const
+{
+  QgsCoordinateReferenceSystem srs;
+
+  static QMutex sMutex;
+  QMutexLocker locker( &sMutex );
+  static QMap<int, QgsCoordinateReferenceSystem> sCrsCache;
+  if ( sCrsCache.contains( mSrid ) )
+  {
+    srs = sCrsCache.value( mSrid );
+    return srs;
+  }
+
+  QgsHanaConnectionRef conn( mUri );
+  srs = conn->getCrs( mSrid );
+
+  if ( srs.isValid() )
+    sCrsCache.insert( mSrid, srs );
+
+  return srs;
+}
+
+QgsVectorLayerExporter::ExportError QgsHanaProvider::createEmptyLayer(
+  const QString &uri,
+  const QgsFields &fields,
+  QgsWkbTypes::Type wkbType,
+  const QgsCoordinateReferenceSystem &srs,
+  bool overwrite,
+  QMap<int, int> *oldToNewAttrIdxMap,
+  QString *errorMessage,
+  const QMap<QString, QVariant> *
+)
+{
+  QgsDataSourceUri dsUri( uri );
+  QgsHanaConnectionRef conn( dsUri );
+  if ( conn.isNull() )
+  {
+    if ( errorMessage )
+      *errorMessage = QObject::tr( "Connection to database failed" );
+    return QgsVectorLayerExporter::ErrConnectionFailed;
+  }
+
+  QString schemaName = dsUri.schema();
+  QString tableName = dsUri.table();
+
+  if ( schemaName.isEmpty() )
+  {
+    if ( errorMessage )
+      *errorMessage = QObject::tr( "Schema name cannot be empty" );
+    return QgsVectorLayerExporter::ErrCreateLayer;
+  }
+
+  QString geometryColumn = dsUri.geometryColumn();
+  QString geometryType;
+
+  QString primaryKey = dsUri.keyColumn();
+  QString primaryKeyType;
+
+  QString schemaTableName = QgsHanaUtils::quotedIdentifier( schemaName ) + '.' +
+                            QgsHanaUtils::quotedIdentifier( tableName );
+
+  if ( wkbType != QgsWkbTypes::NoGeometry && geometryColumn.isEmpty() )
+    geometryColumn = "geom";
```

Comment:
Shouldn't this be `"GEOM"` if the fields are in upper case?

Reply:
Done in  3f7dbe472.